### PR TITLE
feat: Add loading UI in Home

### PR DESCRIPTION
**Pull request by ashrafmahmood**

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -55,10 +55,11 @@ function Home () {
               address={shop.address}
               id={shop.id}
               key={shop.id}
+              loading={loading}
               name={shop.name}
               phone_number={shop.phone_number}
                                   />)))
-          : <p>Loading shops near you...</p>}
+          : (Array(8).fill(null).map(() => <Shop loading={loading}/>))}
```
Comment (Haider8) [inline src/components/Home/Home.js:62]: Why `Array(8)`?

File: src/components/Home/Shop.js [hunk 2]
```diff
@@ -1,17 +1,25 @@
 import { StarIcon } from '@heroicons/react/solid'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Link } from 'react-router-dom'
 
-function Shop ({ name, address }) {
+function Shop ({ address, loading, name }) {
+  const linkDisableHandler = useCallback(
```
Comment (Haider8) [inline src/components/Home/Shop.js:6]: Shouldn't this be called `linkClickHandler` or `shopClickHandler`?

File: src/components/Home/Shop.js [hunk 3]
```diff
@@ -1,17 +1,25 @@
 import { StarIcon } from '@heroicons/react/solid'
-import React from 'react'
+import React, { useCallback } from 'react'
 import { Link } from 'react-router-dom'
 
-function Shop ({ name, address }) {
+function Shop ({ address, loading, name }) {
```
Comment (Haider8) [inline src/components/Home/Shop.js:5]: Please add prop type validation for this component. Like in Auxilliary component.

**Review by Haider8 — CHANGES_REQUESTED** [review]

File: src/components/Home/Shop.js (hunk 2 above)
Comment (ashrafmahmood) [reply src/components/Home/Shop.js:6]: This is for disabling the link.

Comment (Haider8) [reply src/components/Home/Shop.js:7]: Yes but only when in loading state otherwise it is working as usual?

Comment (ashrafmahmood) [reply src/components/Home/Shop.js:7]: Yes

Comment (ashrafmahmood) [reply src/components/Home/Shop.js:7]: It is for disabling the Link when shop component is in loading state

**Pull request merged**